File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    opend= []
    
    for i in s:
        if i == "(":
            opend.append(i)
        else:
            if opend:
                opend.pop()
            else:
                return False
            
    if opend:
        return False
    else:
        return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        int count = 0;
        
        for (char word : s.toCharArray()) {
            if (word == '(') {
                count++;
            }
            else if (word==')') {
                count--;
                if (count <0) {
                    return false;
                }
            }
        }

        return count == 0;
    }
}